client/packages/invoices/src/Returns/CustomerListView/AppBarButtons.tsx

```diff
@@ -68,7 +68,7 @@ export const AppBarButtonsComponent: FC<{
               navigate(
                 RouteBuilder.create(AppRoute.Distribution)
                   .addPart(AppRoute.CustomerReturn)
-                  .addPart(String(customerReturn?.invoiceNumber))
+                  .addPart(String(customerReturn?.id))
                   .build()
               );
```

Comment:
NIT: Noticed that the `try/catch` around `navigate` is mixing two patterns - `async/await` and promises `.then()` chains. We should probably stick to one for consistency (preferably `async/await`, as it's the more modern and widely adopted approach in the JS community).

e.g.

```ts
try {
  const customerReturn = await onCreate({
      id: FnUtils.generateUUID(),
      customerId: name?.id,
      customerReturnLines: [],
  });
  
  if (customerReturn) 
    navigate(
      RouteBuilder.create(AppRoute.Distribution)
        .addPart(AppRoute.CustomerReturn)
        .addPart(String(customerReturn.id))
        .build()
    );
} catch (e) {
  const errorSnack = error(
    `${t('error.failed-to-create-return')} ${(e as Error).message}`
  );
  errorSnack();
}
```

Reply:
Good point, John. I'm just having a read up on those patterns now.

I can see that it is a commonly used mix in our code base (searching via await.*[\n\s].*[\n\s].*[\n\s].*[\n\s].*[\n\s].*then).

I'll make a refactor issue for this